### PR TITLE
*: combine etcd log into dm-master

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

combine etcd log into dm-master

### What is changed and how it works?

- update vendor to support etcd v3.4.3 (for zap log in etcd)
- update golang version to 1.13 (**to compatible with TiDB, PD and etcd**)
- combine etcd log into dm-master

```
go get -u github.com/pingcap/tidb@b274eb2
go get -u github.com/pingcap/tidb-tools@48d5e90
go get -u github.com/pingcap/parser@3b43b46
go get -u github.com/pingcap/pd@2488cb9
go get -u go.etcd.io/etcd@3cf2f69
go mod tidy
```

```
[2019/11/18 14:42:57.180 +08:00] [INFO] [printer.go:54] ["Welcome to dm-master"] ["Release Version"=] ["Git Commit Hash"=64711dd8d39518414123f13a46299b478e4ff72d] ["Git Branch"=etcd-log] ["UTC Build Time"="2019-11-18 06:42:26"] ["Go Version"="go version go1.13.4 darwin/amd64"]
[2019/11/18 14:42:57.180 +08:00] [INFO] [main.go:58] ["dm-master config"="{\"log-level\":\"info\",\"log-file\":\"\",\"log-rotate\":\"day\",\"rpc-timeout\":\"30s\",\"rpc-rate-limit\":10,\"rpc-rate-burst\":40,\"master-addr\":\"127.0.0.1:8261\",\"deploy\":{},\"config-file\":\"dm-master.toml\",\"name\":\"dm-master-XuechengdeMacBook-Pro.local\",\"data-dir\":\"default.dm-master-XuechengdeMacBook-Pro.local\",\"peer-urls\":\"http://127.0.0.1:8291\",\"advertise-peer-urls\":\"http://127.0.0.1:8291\",\"initial-cluster\":\"dm-master-XuechengdeMacBook-Pro.local=http://127.0.0.1:8291\",\"initial-cluster-state\":\"new\",\"join\":\"\"}"]
[2019/11/18 14:42:57.180 +08:00] [INFO] [server.go:103] ["config after join prepared"] [config="{\"log-level\":\"info\",\"log-file\":\"\",\"log-rotate\":\"day\",\"rpc-timeout\":\"30s\",\"rpc-rate-limit\":10,\"rpc-rate-burst\":40,\"master-addr\":\"127.0.0.1:8261\",\"deploy\":{},\"config-file\":\"dm-master.toml\",\"name\":\"dm-master-XuechengdeMacBook-Pro.local\",\"data-dir\":\"default.dm-master-XuechengdeMacBook-Pro.local\",\"peer-urls\":\"http://127.0.0.1:8291\",\"advertise-peer-urls\":\"http://127.0.0.1:8291\",\"initial-cluster\":\"dm-master-XuechengdeMacBook-Pro.local=http://127.0.0.1:8291\",\"initial-cluster-state\":\"new\",\"join\":\"\"}"]
[2019/11/18 14:42:57.180 +08:00] [INFO] [etcd.go:117] ["configuring peer listeners"] [component="embed etcd"] [listen-peer-urls="[http://127.0.0.1:8291]"]
[2019/11/18 14:42:57.181 +08:00] [INFO] [etcd.go:127] ["configuring client listeners"] [component="embed etcd"] [listen-client-urls="[http://127.0.0.1:8261]"]
```

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

### CODE REVIEW

File: pkg/log/log.go [hunk 1]
```diff
@@ -89,7 +90,6 @@ func InitLogger(cfg *Config) error {
 		Level: cfg.Level,
 		File: pclog.FileLogConfig{
 			Filename:   cfg.File,
-			LogRotate:  true,
```
Comment (csuzhangxc) [inline pkg/log/log.go]: `LogRotate` removed in current _pingcap/log_.

**Comment by csuzhangxc** [comment]
@GregoryIan @WangXiangUSTC PTAL

File: dm/master/config.go [hunk 2]
```diff
@@ -324,6 +324,17 @@ func (c *Config) genEmbedEtcdConfig() (*embed.Config, error) {
 	cfg.InitialCluster = c.InitialCluster
 	cfg.ClusterState = c.InitialClusterState
 
+	// use zap as the logger for embed etcd
+	// NOTE: `genEmbedEtcdConfig` can only be called after logger initialized.
+	// NOTE: if using zap logger for etcd, must build it before any concurrent gRPC calls,
+	// otherwise, DATA RACE occur in builder and gRPC.
+	cfg.ZapLoggerBuilder = embed.NewZapCoreLoggerBuilder(log.L().Logger, log.L().Core(), log.Props().Syncer)
```
Comment (WangXiangUSTC) [inline dm/master/config.go]: can we add some tag for etcd's log, just like we add "sarama" in binlog https://github.com/pingcap/tidb-binlog/blob/3f0d148e33777caa1e721000999e84acaebefc28/pkg/util/log.go#L89

Comment (csuzhangxc) [reply dm/master/config.go]: added `[component="embed etcd"]` in https://github.com/pingcap/dm/pull/360/commits/709720aaf03dc57ca8a28a7f28ba38198e5bb7f7.

**Comment by WangXiangUSTC** [comment]
LGTM

File: tests/http_apis/run.sh [hunk 3]
```diff
@@ -31,6 +31,7 @@ function run() {
     task_data=`cat $WORK_DIR/task.yaml.bak`
     rm $WORK_DIR/task.yaml.bak
     echo $task_data
+    sleep 2 # wait for embed HTTP service avaible
```
Comment (GregoryIan) [inline tests/http_apis/run.sh]: can we add a simple backoff function to wait it ok?

File: syncer/db_test.go [hunk 4]
```diff
@@ -275,7 +275,10 @@ func (s *testDBSuite) TestTimezone(c *C) {
 					continue
 				}
 
-				rowid := ev.Rows[0][0].(int32)
+				rowid, ok := ev.Rows[0][0].(int32)
```
Comment (GregoryIan) [inline syncer/db_test.go]: any concurrent case to run? or does not `resetBinlogSyncer` help to filter other cases?

File: tests/http_apis/run.sh (hunk 3 above)
Comment (csuzhangxc) [reply tests/http_apis/run.sh]: added backoff in https://github.com/pingcap/dm/pull/360/commits/aa3d4e77274bb56781582ba0fff369923e4edae2.

File: syncer/db_test.go (hunk 4 above)
Comment (csuzhangxc) [reply syncer/db_test.go]: I revert this change (in https://github.com/pingcap/dm/pull/360/commits/aa3d4e77274bb56781582ba0fff369923e4edae2) and will split the unit test cases (with MySQL) into different containers in CI later.

**Pull request merged**